src/helpers.rs

```diff
@@ -509,3 +509,36 @@ fn bytes_to_os_str<'tcx, 'a>(bytes: &'a [u8]) -> InterpResult<'tcx, &'a OsStr> {
         .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", bytes))?;
     Ok(&OsStr::new(s))
 }
+
+pub fn int_to_immty_checked<'tcx>(
+    int: i128,
+    layout: TyLayout<'tcx>,
+) -> InterpResult<'tcx, ImmTy<'tcx, Tag>> {
+    // If `int` does not fit in `size` bits, we error instead of letting
+    // `ImmTy::from_int` panic.
+    let size = layout.size;
+    let truncated = truncate(int as u128, size);
+    if sign_extend(truncated, size) as i128 != int {
+        throw_unsup_format!(
+            "Signed value {:#x} does not fit in {} bits",
+            int,
+            size.bits()
+        )
+    }
+    Ok(ImmTy::from_int(int, layout))
+}
+
+pub fn uint_to_immty_checked<'tcx>(
+    int: impl Into<u128>,
+    layout: TyLayout<'tcx>,
+) -> InterpResult<'tcx, ImmTy<'tcx, Tag>> {
+    let int = int.into();
+    // If `int` does not fit in `size` bits, we error instead of letting
```

Comment:
Where are we errorring here? :D The code in this function is infallible (except for asserts)

Reply:
I was following the same assertions as `Scalar::from_int` in the same way as `int_to_immty_checked`. iirc it was because @RalfJung didn't want these functions to just panic and just throw a recoverable error instead.

Reply:
I'm not complaining, I'm just saying the return type doesn't need to be fallible and the comment is wrong

Reply:
hahahaha sorry I misunderstood you, yes so that should throw an error instead, let me fix it.